File: phase5testcases/src/test/java/checkoutTest.java
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CheckoutTest {
    private Checkout checkout;

    @BeforeEach
    void setUp() {
        checkout = new Checkout();
    }

    @AfterEach
    void tearDown() {
        checkout = null;
    }

    @Test
    void testSuccessfulCheckout() {
        String result = checkout.processCheckout("1234567891234567", "123", "08/30");
        assertEquals("Successful Checkout", result);
    }

    @Test
    void testInvalidCardNumber() {
        String result = checkout.processCheckout("1234-5678-9123-4567", "123", "08/30");
        assertEquals("Card Number is incorrect", result);
    }

    @Test
    void testInvalidCVV() {
        String result = checkout.processCheckout("1234567891234567", "1234", "08/30");
        assertEquals("CVV is incorrect", result);
    }

    @Test
    void testInvalidExpirationDate() {
        String result = checkout.processCheckout("1234567891234567", "123", "07/24");
        assertEquals("Insert a valid expiration date", result);
    }
}

File: server/Testcases/checkoutTest.js
const Checkout = require('./Checkout');

let checkout;

beforeEach(() => {
  checkout = new Checkout();
});

afterEach(() => {
  checkout = null;
});

test('Successful checkout', () => {
  const result = checkout.processCheckout("1234567891234567", "123", "08/30");
  expect(result).toBe("Successful Checkout");
});

test('Invalid card number', () => {
  const result = checkout.processCheckout("1234-5678-9123-4567", "123", "08/30");
  expect(result).toBe("Card Number is incorrect");
});

test('Invalid CVV', () => {
  const result = checkout.processCheckout("1234567891234567", "1234", "08/30");
  expect(result).toBe("CVV is incorrect");
});

test('Invalid expiration date', () => {
  const result = checkout.processCheckout("1234567891234567", "123", "07/24");
  expect(result).toBe("Insert a valid expiration date");
});
